### PR TITLE
Initial setup and add helloworld example

### PR DESCRIPTION
**Pull request by sklam**
- setup basic structure for the repo
- add setup.py
- precommit with `black`
- github action for testing and building conda-packages
   - we can setup pypi publishing in another PR.

### CODE REVIEW

File: .github/workflows/test_and_deploy.yml [hunk 1]
```diff
@@ -0,0 +1,99 @@
...
+        run: |
+          source "$CONDA/etc/profile.d/conda.sh"
+          conda activate ci
+          pytest numba_extras
+      - name: Conda Build
+        run: |
+          source "$CONDA/etc/profile.d/conda.sh"
+          conda build --python ${{ env.MAIN_PYVER }} -c defaults -c numba -c conda-forge buildscripts/conda
+          # This doesn't rebuild, but simply computes the name of the file that was previously built
+          OUTPUT=$(conda build --output --python ${{ env.MAIN_PYVER }} -c defaults -c numba -c conda-forge buildscripts/conda)
```
Comment (stuartarchibald) [inline .github/workflows/test_and_deploy.yml:49]: Is conda-forge needed in here? I guess yes for `black` versions?

File: .github/workflows/test_and_deploy.yml [hunk 2]
```diff
@@ -0,0 +1,99 @@
...
+          # Setup a local channel
+          BUILT_PKG=`ls ./artifact_storage | head -1`
+          mkdir -p my_channel/noarch
+          mv ./artifact_storage/$BUILT_PKG my_channel/noarch/
+          cd my_channel
+          conda index .
+          cd ..
+          # Install artifact
+          conda install -q -c ./my_channel -c numba numba-extras
+          conda list
```
Comment (stuartarchibald) [inline .github/workflows/test_and_deploy.yml:94]: Suggest running `numba -s` somewhere in here?

File: pyproject.toml [hunk 3]
```diff
@@ -0,0 +1,23 @@
+[tool.black]
+line-length = 88
+target-version = ['py37']
```
Comment (stuartarchibald) [inline pyproject.toml:3]: main build version is declared as 3.8 in the above?

**Review by stuartarchibald — COMMENTED** [review]
Thanks for setting this up @sklam.

I've reviewed the patches and have also:

* Checked this out and built it locally.
* Tested my local build with `pytest`.
* Tested that the pre-commit config and auto-formatter works.

File: .github/workflows/test_and_deploy.yml (hunk 1 above)
Comment (sklam) [reply .github/workflows/test_and_deploy.yml:49]: yes, for `black`

File: .github/workflows/test_and_deploy.yml (hunk 2 above)
Comment (sklam) [reply .github/workflows/test_and_deploy.yml:94]: yup, good idea. i'll add.

File: pyproject.toml (hunk 3 above)
Comment (sklam) [reply pyproject.toml:3]: i don't think it matters that the build is done on py3.8 but we ask `black` to use py3.7. The two are separate concerns.

Comment (stuartarchibald) [reply pyproject.toml:3]: ok, wasn't sure if `black` minded.

File: .github/workflows/test_and_deploy.yml (hunk 2 above)
Comment (stuartarchibald) [reply .github/workflows/test_and_deploy.yml:94]: thanks

Comment (sklam) [reply .github/workflows/test_and_deploy.yml:94]: added in https://github.com/numba/numba-extras/pull/1/commits/efde256a06a2a354af713b44c0052d160a352288

Comment (sklam) [reply .github/workflows/test_and_deploy.yml:94]: https://github.com/numba/numba-extras/pull/1/checks?check_run_id=1500309445#step:5:10

Comment (sklam) [reply .github/workflows/test_and_deploy.yml:94]: much better linking than azure

Comment (stuartarchibald) [reply .github/workflows/test_and_deploy.yml:94]: agree

**Review by stuartarchibald — APPROVED** [review]
Thanks for setting all this up, looks good!

**Pull request merged**